module-2/backend/week_four.md

```diff
@@ -10,14 +10,40 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+small amnt of data stored on your local machine (usually consists of a user id), which tells the server that you have been to the site before
+
 * What’s the difference between a session and a cookie?
+sessions are encrypted cookies
+
 * What’s a flash and when do you want to use flashes?
+a small notice which allows you to notify the user, on the webpage, exactly what is/is not happening in the background - they are not permanent - you want a flash if someone creates something/updates something, etc.
+
 * Why do people say “HTTP is stateless”?
+because the sessions, and associated info, lies with the client, NOT the server
+
 * What’s authentication? Explain.
+authentication is allowing access to some/all pages of a website
```

Comment:
More generally authentication is keeping track of who a user is, as opposed to just responding to all requests as anonymous users.